src/main/java/picard/sam/SamToFastq.java

```diff
@@ -285,7 +290,7 @@ private File makeReadGroupFile(final SAMReadGroupRecord readGroup, final String
         }
         fileName = IOUtil.makeFileNameSafe(fileName);
         if (preExtSuffix != null) fileName += preExtSuffix;
-        fileName += ".fastq";
+        fileName += COMPRESS_OUTPUTS_PER_RG ? ".fastq.gz" : ".fastq";

```

Comment:
My thought was that you'd add "USE_COMPRESSED_OUTPUT" and that will determine the file suffixes.